docs: Add roadmap, update READMEs and contributing guide (#295)
dylibso/chicory: README.md

### Why is this needed?

> If you'd prefer to watch a video interested of read, see our [2024 Wasm I/O](https://2024.wasmio.tech/) on the subject:

Author: ```suggestion
If you'd prefer to watch a video instead of reading, see our [2024 Wasm I/O](https://2024.wasmio.tech/) on the subject:
```

### Roadmap

Chicory development was started in September, 2023. The following are the milestones we're aiming for. These
are subject to change but represent our best guesses with current information. These are not necessarily sequential
and some may be happening in parallel. Unless specified, any unchecked box is still not planned or started.
If you have an interest in working on any of these please reach out in Zulip!

#### Bootstrap a bytecode interpreter and test suite (EOY 2023)

* [x] Wasm binary parser [link](wasm/)
* [x] Simple bytecode interpreter
* [x] Establish basic coding and testing patterns
* [x] Generate JUnit tests from wasm test suite [link](test-gen-plugin/)

#### Make the interpreter production ready (Summer 2024)

* [ ] Make all tests green with the interpreter (important for correctness)
  * Almost complete
* [ ] Implement validation logic (important for safety)
  * Started by [@andreaTP](https://github.com/andreatp)
* [ ] Draft of the v1.0 API (important for stability and dx)

#### Make it fast (EOY 2024)

The primary goal here is to create an AOT compiler that generates JVM bytecode
as interpreting bytecode can only be so fast.

> * [ ] Decouple interpreter and create separate compiler and interpreter "engines"

Author: ```suggestion
* [ ] Decouple interpreter and create separate compiler and interpreter "engines"
  * Started by @danielperano
```

* [ ] Proof of concept AOT compiler (run some subset of modules)
* [ ] AOT engine passes all the same specs as interpreter (stretch goal)
* [ ] Off-heap linear memory (stretch goal)

### Make it compatible (EOY 2024)

* [ ] WASIp1 Support (including test gen)
  * We have [partial support for wasip1](wasi/) and [test generation](wasi-test-gen-plugin/)
* [ ] SIMD Support

>   * Started as a GSoC project by [@zedbeit](https://github.com/zedbeit)

Author: ```suggestion
  * Started by [@zedbeit](https://github.com/zedbeit)
```
